hoomd/operations.py

```diff
@@ -117,20 +197,50 @@ def integrator(self, op):
 
     @property
     def updaters(self):
+        """list[``Updater``]: A list of updater operations.
+
+        Holds the list of updaters associated with this object. The list can be
+        motified as a standard Python list.
+        """
         return self._updaters
 
     @property
     def analyzers(self):
+        """list[``Analzyer``]: A list of updater operations.
+
+        Holds the list of updaters associated with this object. The list can be
+        motified as a standard Python list.
+        """
         return self._analyzers
 
     @property
     def tuners(self):
+        """list[``Tuner``]: A list of updater operations.
+
+        Holds the list of updaters associated with this object. The list can be
+        motified as a standard Python list.
+        """
         return self._tuners
 
     def __iadd__(self, operation):
+        """Works the same as `Operations.add`.
+
+        Args:
+            operation (``operation``): A HOOMD-blue updater, analyzers, compute,
+                tuner, or integrator to add to the object.
+        """
         self.add(operation)
+        return self
 
     def remove(self, operation):
+        """Remove an operation from the `Operations` object.
+
+        Checks for removal according to object id.
+
+        Args:
+            operation (``operation``): A HOOMD-blue updater, analyzers, compute,
+                tuner, or integrator to remove from the object.
+        """
         if isinstance(operation, hoomd.integrate._BaseIntegrator):
             raise ValueError(
                 "Cannot remove iterator without setting to a new integator.")
```

Comment:
This method needs to raise a `TypeError` after the `elif`s if the argument is not an operation class!

Reply:
Do we need to error here? It for sure makes sense to error in `add`, but here I think it makes sense that nothing happens if an user calls `operations.remove(3)` since `3 not in operations`.

Reply:
Trying to remove an object that doesn't exist in the collection typically raises an exception. In the case where none of the `isinstance` checks pass, I would expect a `TypeError` instead of a `ValueError`. However, I would expect a `ValueError` if one tries to remove an updater/tuner/analyzer that isn't actually in this `Operations` container.
```pytb
>>> data = [1, 2, 3]
>>> data.remove('potato')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: list.remove(x): x not in list
```

Reply:
Why can't you remove an integrator without setting a new integrator? The default integrator is None in Python (a NULL shared pointer in C++).

Reply:
We could do that then. I didn't know we were accepting that as a valid state.

Reply:
For no other reason than it is a valid default state, so we should allow users to return the object to its default state after modifying it.

Reply:
I agree and have modified it accordingly. I meant I didn't know it was a valid C++ state.